### PR TITLE
Update dedupe/dependencies and remove obsolete temporary fixes

### LINKED ISSUE #201: Remove manually overridden patch.crates-io dependencies
The comments above them indicate that basically all of them are no longer relevant, so we probably dont' want to be using old versions of these

### LINKED ISSUE #218: Update/Dedupe dependency versions
We're a few versions back on several dependencies (URL, reqwest, ...). We should probably do this before/with #201.

### PR DESCRIPTION
**Pull request by eoger**
Fixes #201 and fixes #218.
Note that we do a bunch of `HeaderMap::get().and_then()[...]` which eats a bunch (UTF-8 parsing/number parsing) of errors, do we want to catch all of them?

### CODE REVIEW

File: fxa-client/Cargo.toml [hunk 1]
```diff
@@ -4,23 +4,23 @@ version = "0.1.0"
...
-hex = "0.3.1"
+hex = "0.3.2"
 lazy_static = "1.0.0"
-log = "0.4"
-openssl = { version = "0.10.7", optional = true }
+log = "0.4.5"
+openssl = { version = "0.10.12", optional = true }
 regex = "1.0.0"
-reqwest = "0.8.2"
-ring = "0.13.0-alpha5"
```
Comment (vladikoff) [inline fxa-client/Cargo.toml]: wow stable ring!

**Comment by rfk** [comment]
> do we want to catch all of them?

Great, now I'm singing the Pokémon theme-song in my head...

**Comment by eoger** [comment]
By the way big thanks to @seanmonstar for his precious assistance on the `reqwest` upgrade.

File: sync15-adapter/src/token.rs [hunk 2]
```diff
@@ -78,7 +79,8 @@ impl TokenFetcher for TokenServerFetcher {
             debug!("  Response body {}", resp.text().unwrap_or_else(|_| "???".into()));
             // XXX - shouldn't we "chain" these errors - ie, a BackoffError could
             // have a TokenserverHttpError as its cause?
-            if let Some(ms) = resp.headers().get::<RetryAfter>().map(|h| (**h * 1000f64) as u64) {
+            // XXX - we are silentely dropping UTF8 parsing/number parsing errors here.
+            if let Some(ms) = resp.headers().get(RETRY_AFTER).and_then(|v| v.to_str().ok()).and_then(|s| s.parse::<f64>().ok()).map(|f| (f * 1000f64) as u64) {
```
Comment (thomcc) [inline sync15-adapter/src/token.rs]: We should see if we can enter this block if there's a retry after header at all. Maybe make up a backoff time if we fail to get a reasonable one? Doesn't seem good to silently ignore it.

**Pull request merged**